src/variables.jl

```diff
@@ -14,10 +14,52 @@ mutable struct VariableInfo{S, T, U, V}
     fixedvalue::U
     hasstart::Bool
     start::V
-    binary::Bool
-    integer::Bool
+    binary::BT
+    integer::IT
 end
 
+function setlowerbound(info::VariableInfo, lower, _error::Function)
+    info.haslb && _error("Cannot specify variable lowerbound twice")
+    info.haslb = true
+    info.lowerbound = lower
+    # lowerbound might be of different type than info.lowerbound
```

Comment:
This is pretty awkward. Why not use a struct with `Any` members for the processing that happens at macro time, and then turn it into a strongly-typed struct for the processing that happens at runtime?

Reply:
Good idea!